internal/throttler/dispatch_throttler.go

```diff
@@ -0,0 +1,103 @@
+package throttler
+
+import (
+	"context"
+	"github.com/openfga/openfga/internal/build"
+	"github.com/openfga/openfga/pkg/telemetry"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"time"
+)
+
+// DispatchThrottlingConfig encapsulates configuration for dispatch throttling check resolver
+type DispatchThrottlingConfig struct {
+	Frequency time.Duration
+	Threshold uint32
+}
+
+type Ticker interface {
+}
+
+// DispatchThrottler will prioritize requests with fewer dispatches over
+// requests with more dispatches.
+// Initially, request's dispatches will not be throttled and will be processed
+// immediately. When the number of request dispatches is above the Threshold, the dispatches are placed
+// in the throttling queue. One item form the throttling queue will be processed ticker.
+// This allows a check / list objects request to be gradually throttled.
+type DispatchThrottler struct {
+	config          DispatchThrottlingConfig
+	ticker          *time.Ticker
+	throttlingQueue chan struct{}
+	done            chan struct{}
+}
+
```

Comment:
```suggestion
// NewDispatchThrottler constructs a DispatchThrottler which can be used to control the rate of dispatched subproblems in FGA queries.
```

Please add this Godoc, and in general let's strive for godoc on any public signature of a package (even `internal` public signatures).